File: Languages/dotnet/src/model/SimpleTableAnalysis.cs
using System.Collections.Generic;

namespace inputclass.src.model {
    /// <summary>
    /// Interface "SimpleTableAnalysis"
    /// </summary>
    public interface SimpleTableAnalysis {      

        /// <summary>
        /// Método de interface, responsável por realizar a análise na lista de "UserInfo"
        /// </summary>
        /// <param name="list">Lista de "UserInfo"</param>
        /// <returns>valor da análise</returns>
        double analysis(List<UserInfo> list);
    }
}

File: Languages/scala/src/model/SimpleTableAnalysis.scala
package model

/**
 * Interface para analise de uma lista de usuarios
 */
trait SimpleTableAnalysis {

  /**
   * Método de interface, responsável por realizar uma analise em uma lista de usuarios
   * @param userInfoList Lista de usuarios
   * @return resultado da analise
   */
  def analysis(userInfoList: List[UserInfo]): Double
}
